BZ 858997 - Fix handling of unauthenticated XHR requests

n1zyy opened the pull request:
Both the login page and the rest of the app now support handling
an HTTP 401 response when the request is unauthenticated.

https://bugzilla.redhat.com/show_bug.cgi?id=858997

Commit:
BZ 858997 - Fix handling of unauthenticated XHR requests

Both the login page and the rest of the app now support handling
an HTTP 401 response when the request is unauthenticated.

https://bugzilla.redhat.com/show_bug.cgi?id=858997

jtomasek commented on src/app/views/user_sessions/unauthenticated.js.erb:
```diff
@@ -1 +0,0 @@
-$("#notifications").html("<%= escape_javascript(render :partial => "notifications") %>");
```
This file could get deleted as it is empty.

n1zyy replied:
I believe it is deleted and GitHub is just not showing it clearly. The file no longer exists in the branch I pushed.

jtomasek commented:
ACK, I think this ended up as quite nice solution. Works as expected. Some nits inline.

jtomasek commented on src/app/views/user_sessions/new.html.haml:
```diff
@@ -10,6 +10,11 @@
+    $("#login_form").bind("ajax:error", function(event, data, status, xhr) {
+      $("#notifications").html("<div class=\'error\'>\n <p class=\'text\'>#{t("user_sessions.flash.warning.login_failed")}<\/p>\n<\/div>\n");
+      $('form .spinner').fadeOut('fast');
+    });
```
This should be surrounded by `$(document).ready(function(){ ... });` just to be safe.

Commit:
BZ 858997 - Wraps login page JS with document.ready

The code to handle 401s on the login page should only be called
when the document is ready.

jtomasek commented:
ACK
